linetools/lists/linelist.py

```diff
@@ -460,23 +476,40 @@ def sortdata(self, keys, reverse=False):
         if keys[0] == 'as_given':
             return
 
+        # hstack for convenience
+        if len(self._extra_table) == len(self._data):
+            flg_extra = True
+            dkeys = self._data.keys()
+            ekeys = self._extra_table.keys()
+            dtbl = hstack([self._data, self._extra_table], join_type='exact')
+        else:
+            flg_extra = False
+            dtbl = self._data
+
         # sort
-        self._data.sort(keys)
+        dtbl.sort(keys)
 
         # reverse?
         if reverse:
-            self._data.reverse()
+            dtbl.reverse()
         self.sort_by = keys
 
+        # Finish
+        if flg_extra:
+            self._data = dtbl[dkeys]
+            self._extra_table = dtbl[ekeys]
```

Comment:
`ion_name` does not exist in `self._data` right?

Reply:
That is correct.